src/util/filesystem.cxx

```diff
@@ -0,0 +1,9 @@
+#include "filesystem.hxx"
+
+std::string fs::getBasePath() noexcept
+{
+  char * path = SDL_GetBasePath();
```

Comment:
In case of error, this will be NULL. Creating the std::string from NULL will result in exception. Please handle the error case here.